aion-etl/src/main/java/aion/dashboard/stats/CirculatingSupplyTask.java

```diff
@@ -0,0 +1,50 @@
+package aion.dashboard.stats;
+
+import aion.dashboard.service.*;
+import aion.dashboard.util.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class CirculatingSupplyTask implements Runnable {
+    private DBService dbService;
+    private static final Logger GENERAL = LoggerFactory.getLogger("logger_general");
+    private ExecutorService executorService = Executors.newSingleThreadExecutor();
+
+    public CirculatingSupplyTask(DBService dbService) {
+        this.dbService=dbService;
+
+    }
+
+    public CirculatingSupplyTask(){
+        dbService=DBService.getInstance();
+    }
+    public void start(){
+        executorService.submit(this);
+    }
+
+    public void stop(){
+        executorService.shutdownNow();
+    }
+    @Override
+    public void run(){
+
+        do {
+            try {
+                Thread.currentThread().setName("circulating supply task");
+
+                if (!Thread.currentThread().isInterrupted()) {
+                    dbService.generateCirculatingSupply();
+                }
+            }catch (Exception e){
+                GENERAL.warn("Error: ", e);
+            }
+// run every hour
+        }while (Utils.trySleep(60000*60));
```

Comment:
A suggestion, you might want to take a look at: https://github.com/quartz-scheduler/quartz/ for enterprise level cron-like scheduling for future improvements.

Reply:
Great, I did not use any other method for schedule, to keep the current format and have two type